airflow/config_templates/config.yml

```diff
@@ -1230,6 +1230,14 @@
       type: string
       example: ~
       default: "30"
+    - name: removed_tasks_lead_to_dagrun_running
+      description: |
+        Do we keep DagRun "running" when it has a task in the "removed" state? (Tasks are marked as
```

Comment:
Didn't a previous version of this talk about marking the dag run as failed?

Leaving the dagrun in the running state seems like it would be running forever which doesn't sound like what we want to happen.

Reply:
running (rather than failed) is what I've used in my patch in prod (for few months) to resolve it as eventually the tasks get into proper state on dynamic dag (keeping dagrun as 'running' is better than setting to 'failed' if task state is going into 'removed' on transient basis)

Reply:
Eek! This is very very different behaviour to what we discussed when this PR was first opened.

For most people and almost every other case this behaviour will be **very** wrong. If your dynamic dags are "randomly" removing tasks the fix is to not make them do that. Do you not get random task failures too as a result of that? (Where Airflow tries to run a task but by the time the executor starts the dag doesn't contain that dag)

Is the current behaviour of Airflow such that (so long as the dagrun is still running) that when those tasks come back that the scheduler still picks them up?

If, and this is a big if, we keep this change as it is this config option should be called `dagrun_stays_running_with_removed_tasks`, and we should make it clear in the docs that without something to add-back the missing dags, or a dagrun_timeout this will lead to dagruns that never complete.